src/main/java/seedu/duke/commands/DeleteCommand.java

```diff
@@ -0,0 +1,58 @@
+package seedu.duke.commands;
+
+import seedu.duke.exceptions.NoSuchTaskException;
+import seedu.duke.tasks.Module;
+import seedu.duke.tasks.ModuleList;
+import seedu.duke.tasks.Task;
+import seedu.duke.tasks.TaskList;
+
+public class DeleteCommand extends Command {
+
+    private String moduleCode = "";
+    private int taskNumber = -1;
+    private String result = "";
+
+    public DeleteCommand(String moduleCode) {
+        this.moduleCode = moduleCode;
+    }
+
+    public DeleteCommand(String moduleCode, int taskNumber) {
+        this.moduleCode = moduleCode;
+        this.taskNumber = taskNumber;
+    }
+
+    @Override
+    public CommandResult execute(ModuleList moduleList) {
+        if (taskNumber <= 0) {
+            deleteModule(moduleList);
+        } else if (moduleCode.isBlank()) {
+            deleteTask(moduleList);
+        } else {
+            deleteTaskFromModule();
+        }
+        return new CommandResult(result);
+    }
+
+    public void deleteModule(ModuleList moduleList) {
+        moduleList.removeModule(moduleCode);
+        result = moduleCode + " has been deleted.";
```

Comment:
This part could be simplified if `removeModule` returned the module that was deleted instead of being a `void` method. Then you could just do `result = moduleList.removeModule(moduleCode)` + " has been deleted".

Also, maybe convert the string constant into a format string (`"%s has been deleted."`) and call String.format here instead.